### PR TITLE
73 hygen

### LINKED ISSUE #73: Implement Hygen for component creation

### PR DESCRIPTION
**Pull request by Incin**

### CODE REVIEW

**Review by dpenciso — APPROVED** [review]
:)

File: .hygen/new/component/component.tsx.ejs.t [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+const <%= name %>: React.FC<<%= name %>Props> = () => {
+  return (
+    <div className={styles.<%= h.changeCase.lcFirst(name) %>}>
+      <%= name %>
+    </div>
+  );
+};
+
+export default <%= name %>;
```
Comment (jefferymills) [inline .hygen/new/component/component.tsx.ejs.t:21]: What do you think of adding a propTypes entry? Seems like most will need it ya?

File: .hygen/new/component/test.js.ejs.t [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  it('should render', () => {
+    render(
+      <Provider store={store}>
+        <<%= name %>/>
+      </Provider>
+    );
+
+    expect(screen.getByText('<%= name %>')).toBeInTheDocument();
+  });
+});
```
Comment (jefferymills) [inline .hygen/new/component/test.js.ejs.t:18]: I'm thinking that if it's a true unit test that we wouldn't be mocking the store or wrapping in a Provider. Thoughts?

Comment (Incin) [reply .hygen/new/component/test.js.ejs.t:18]: not sure, i just copied the one dave wrote with you and put it into a template, im fixing it now

File: .hygen/new/component/component.tsx.ejs.t (hunk 1 above)
Comment (Incin) [reply .hygen/new/component/component.tsx.ejs.t:21]: ah yes we need that

File: .hygen/new/component/test.js.ejs.t (hunk 2 above)
Comment (jefferymills) [reply .hygen/new/component/test.js.ejs.t:18]: Ok, I would say remove anything to do with redux/store then. I think we'll only want to do unit tests for dumb components.

**Review by jefferymills — APPROVED** [review]
🇬🇵

**Pull request merged**